### PR TITLE
feat: update pre-relase reusable workflow to support nrjmx based integrations

### PR DESCRIPTION
**Pull request by sigilioso**
Introduce the parameters required for nrjmx based integrations in the reusable pre-release workflow (in the same way it was done for push/pr fake-prerelease #51)

### CODE REVIEW

File: .github/workflows/reusable_pre_release.yaml [hunk 1]
```diff
@@ -44,6 +44,17 @@ on:
         default: msi
         required: false
         type: string
+      goarch_matrix:
+        description: "Defines a custom goarch-matrix for windows builds as nrjmx based integrations support amd64 only"
```
Comment (kang-makes) [inline .github/workflows/reusable_pre_release.yaml:48]: Usually actions take the arguments as comma-separated list.

NIT: Leave a comment stating that it needs to be in JSON format

Comment (kang-makes) [reply .github/workflows/reusable_pre_release.yaml:48]: If it iw for Windows only, I would also change the name of the argument from `goarch_matrix` to `windows_build_goarch_matrix`.

**Review by kang-makes — APPROVED** [review]
A couple of NITs but if LGTM.

File: .github/workflows/reusable_pre_release.yaml (hunk 1 above)
Comment (sigilioso) [reply .github/workflows/reusable_pre_release.yaml]: I've added the comment and updated the names, thanks! 🙂

**Pull request merged**